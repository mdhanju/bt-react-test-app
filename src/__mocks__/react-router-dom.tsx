/* eslint-disable */

import React, { ReactNode } from 'react';

type Props = {
    children: ReactNode;
};

const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: Props): ReactNode => <div>{children}</div>;
module.exports = rrd;
