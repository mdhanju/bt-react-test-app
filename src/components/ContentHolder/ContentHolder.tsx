import React, { ReactNode } from 'react'
import { Container, Row, Col } from 'reactstrap'

import Menu from '../Menu'

const ContentHolder = ({ children }: Props): React.ReactElement<Props> => {
    return (
        <Container fluid className="d-flex flex-column">
            <Menu />

            <Row className="h-100 align-self-center">
                <Col className="my-auto">{children}</Col>
            </Row>
        </Container>
    )
}

type Props = {
    children?: ReactNode;
};

export default ContentHolder
