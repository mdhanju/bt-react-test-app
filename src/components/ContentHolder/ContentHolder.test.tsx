import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import ContentHolder from '../ContentHolder'
import { Container, Row } from 'reactstrap'

describe('<ContentHolder />', () => {
    it('should render initial layout', () => {
        const wrapper = shallow(<ContentHolder />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should render child component', () => {
        const wrapper = shallow(
            <ContentHolder>
                <div className="spinner-border" />
            </ContentHolder>
        )
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.find(Container)).toHaveLength(1)
        expect(wrapper.find(Row)).toHaveLength(1)
        expect(wrapper.contains(<div className="spinner-border" />)).toBe(true)
    })

    it('should render Div with Text', () => {
        const wrapper = render(
            <MemoryRouter>
                <ContentHolder>
                    <span>Test Name</span>
                </ContentHolder>
            </MemoryRouter>
        )
        expect(wrapper.text()).toBe('Home - Sample FormTest Name')
        expect(wrapper.html()).toContain('div')
        expect(wrapper.html()).toContain('span')
    })
})
