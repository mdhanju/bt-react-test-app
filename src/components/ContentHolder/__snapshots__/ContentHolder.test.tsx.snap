// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHolder /> should render initial layout 1`] = `
Array [
  <Container
    className="d-flex flex-column"
    fluid={true}
    tag="div"
  >
    <Menu />
    <Row
      className="h-100 align-self-center"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        className="my-auto"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      />
    </Row>
  </Container>,
]
`;
